aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncTime.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+import androidx.annotation.Nullable;
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.util.Time;
+
+/**
+ * Represents the time at which a type of model was last sync'd.
+ * Why isn't that just an @Nullable Long (capital 'L')?
+ * Rx doesn't allow nulls, so we have to encapsulate the Long value into this
+ * class. TODO: move this to become a static inner class of {@link LastSyncMetadata}?
+ */
+final class SyncTime {
```

Comment:
Ooof, the fact we have to have this class due to RX hurts my eyes - I like the idea of moving it to a static inner class.

Reply:
Also, do we still need Rx now that we're requiring Java 1.8 compatability sections be added to the code in the docs?